examples/react-full-v1/pages/star-wars/@id/+onBeforeRender.tsx

```diff
@@ -1,15 +1,16 @@
 // https://vike.dev/onBeforeRender
-export default onBeforeRender
 
 import fetch from 'cross-fetch'
 import { filterMovieData } from '../filterMovieData'
-import type { PageContextBuiltInServer } from 'vike/types'
+import type { OnBeforeRender, OnBeforeRenderReturn, PageContextClient, PageContextServer } from 'vike/types'
 import type { MovieDetails } from '../types'
 import { render } from 'vike/abort'
 import React from 'react'
 
-async function onBeforeRender(pageContext: PageContextBuiltInServer) {
-  const dataUrl = `https://star-wars.brillout.com/api/films/${pageContext.routeParams.id}.json`
+const onBeforeRender: OnBeforeRender = async (
+  pageContext: PageContextServer | PageContextClient
+): OnBeforeRenderReturn => {
```

Comment:
@brillout is this what you had in mind? If yes I'd go ahead and make change all the hooks to this pattern

Reply:
Yes, although I'm still a little bit hesitating between `pageContext: PageContextServer | PageContextClient` and  `pageContext: OnBeforeRenderParam`.

What do you think? 👀

Reply:
Let me think a little bit more about it; I'll circle back on this tentatively later today.

> I'm not a big fan of the entire `OnBeforeRender*` set of types

(We can further discuss this if you want, as you wish.)

Reply:
Alright, I'm very happy with 7bc32ed79c5a50f5b081708a26fd1a5265c58ae6. While we tell users:

> We recommend to always use `ReturnType<OnBeforeRender>` which is needed for making TypeScript strict about return types, see [some-ts-playground-link] and [some-external-link-explaining-it].

I feel like this nails it. 

As for the expanded styles, users can do it if they want to although, IMO, I wouldn't encourage it nor mention it in the docs: I think IntelliSense's QuickInfo is more than enough.

![Screenshot 2023-10-10 11 49 25](https://github.com/vikejs/vike/assets/1005638/8da3f793-d9c0-47f5-8a03-a1568d25c32c)

![image](https://github.com/vikejs/vike/assets/1005638/f37c825e-f35e-4e7a-a050-e00878155a6e)

Let me know if you have objections (I'm more than open to further discuss things if you us want to).

Reply:
Ah well done, this looks great now. Doesn't feel clunky anymore. Let me go ahead and apply that to all hooks

Reply:
👍 I'm glad you like it as well.

Let me know if repeating the JSDocs gets too annoying, I think we can write a little macro script and append it to `vike/dist-cjs-fixup.mjs` (I guess we should rename it to `vike/build-fixup.js` then).

Reply:
It might be a bit of an overkill, especially because the duplication is not spread across different files